Converts ALL typepaths to lowercase

nemvar opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

From now on all paths are required to use snake case.
Replaces all capital letters in typepaths with lowercase ones. Converts CamelCase to snake_case. Renames some paths that are hard to read.
Changes
``/obj/item/reagent_container/pill/C2`` to ``/obj/item/reagent_container/pill``
and 
``/obj/item/storage/pill_bottle/C2`` to ``/obj/item/storage/pill_bottle``
since those paths don't do anything.

Regex used to find these: ``
(?<![^\n])/((mob|datum|atom|obj|area)([^\s^\(^/]|/(?!(/|var/|proc/|\*|#)))*[A-Z])(?!\S*\()``
Not aware of comments. Automatic detection should probably happen with spacemanDMM, no idea how that works though. Will try my best.

## Why It's Good For The Game

It's in the guidelines.
conflicts with https://github.com/tgstation/tgstation/pull/51634

nemvar committed:
Case of lower

nemvar committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into lowercase

nemvar committed:
More changes

nemvar committed:
Ruins the nice 420 diff, brainfart when doing the second batch of conversions

Fikou commented:
DONT CHANGE IAN

nemvar committed:
More changes

nemvar committed:
Next batch. I think

nemvar committed:
Converts even more paths

nemvar committed:
Restarts bots

nemvar committed:
Capital Free Zone

nemvar committed:
Come on travis, do something

TerraGS commented:
My favorite kind of code improvement next to performance boosts.

Rohesie commented:
Blessed.

ShizCalev commented on code/game/objects/items/storage/firstaid.dm:
```diff
@@ -378,13 +378,13 @@
 		new /obj/item/reagent_containers/pill/potassiodide(src)
 
-/obj/item/storage/pill_bottle/C2/probital
+/obj/item/storage/pill_bottle/probital
```
Undocumented

nemvar replied:
Undocumented but no unintentional. Change does not alter any behaviour.

optimumtact commented:
If we're going to standardise we should have lints that enforce it and some documentation

MrPerson commented:
I notice your regex is missing areas.

nemvar committed:
Renames areas

nemvar committed:
Bots, please stop dying

nemvar commented:
I'm working on the linting but what exactly do you want documented? There is a single actual change in here which is removing the /C2 paths.

ATH1909 commented:
based

ATH1909 commented:
make sure that the DisIntegrate() proc (it's swarmer-related, IIRC) doesn't break because of this PR

optimumtact commented:
making it clear snake case is the required method naming style

ShizCalev commented on code/modules/vending/modularpc.dm:
```diff
@@ -19,7 +19,7 @@
 		           /obj/item/computer_hardware/ai_slot = 2,
 		           /obj/item/computer_hardware/printer/mini = 2,
-		           /obj/item/computer_hardware/recharger/APC = 2,
+		           /obj/item/computer_hardware/recharger/apc = 2,
```
if you wouldn't mind, give this a slightly longer name while you're at it. apc_recharger or something like that would be awesome.

spookydonut reviewed (changes requested):
why not add your detection regex to `/tools/travis/check_grep.sh`

nemvar committed:
Updates CONTRIBUTING.md and updates a few paths I missed.

nemvar committed:
APC recgarftzfvas
/obj/item/computer_hardware/recharger/apc to /obj/item/computer_hardware/recharger/apc_recharger

ShizCalev commented:
@MrStonedOne @Cyberboss need feedback to ensure sql and tgs3/4 don't get fucked up by this since SQL's unit test keeps failing (ie changing DBQuery to db_query )

ShizCalev commented:
@MrStonedOne @Cyberboss @SpaceManiac  might need feedback to ensure sql / rust_g.dll and tgs3/4 don't get fucked up by this since mySQL's unit test keeps failing (ie changing DBQuery to db_query )

MrStonedOne commented:
todo: update server_side_modifications.dm on the game servers. (oranges, this should now be setup to allow editing with admin rights)

SpaceManiac commented:
rust-g should be fine as its `rust_g.dm` isn't touched. I don't know anything about how TGS or Postgres relate to this

Cyberboss commented:
TGS uses snake case, so it should be fine. I don't see it in here.

optimumtact commented:
Approving PR's does nothing, merging them is the only valuable feedback

spookydonut commented:
@Rohesie @optimumtact didn't have the regex added to check_grep.sh, shouldnt have been merged

SpaceManiac commented:
> didn't have the regex added to check_grep.sh, shouldnt have been merged

not a blocker IMO, certainly a nice-to-have tho

optimumtact commented:
@spookydonut if you dont' want your PR merged, mark it as draft
